src/gui/gui_modifier_tooltips.cpp

```diff
@@ -50,41 +50,185 @@ void modifier_description(sys::state& state, text::layout_base& layout, dcon::mo
 		text::add_to_layout_box(layout, state, box, std::string_view{":"}, text::text_color::white);
 		text::add_space_to_layout_box(layout, state, box);
 		auto color = data.positive_is_green
-		                 ? (prov_def.values[i] > 0.f ? text::text_color::green : text::text_color::red)
-		                 : (prov_def.values[i] > 0.f ? text::text_color::red : text::text_color::green);
+		                 ? (prov_def.values[i] >= 0.f ? text::text_color::green : text::text_color::red)
+		                 : (prov_def.values[i] >= 0.f ? text::text_color::red : text::text_color::green);
 		text::add_to_layout_box(layout, state, box, format_modifier_value(state, prov_def.values[i], data.type), color);
 		text::close_layout_box(layout, box);
 	}
 
 	const auto& nat_def = fat_id.get_national_values();
 	for(uint32_t i = 0; i < nat_def.modifier_definition_size; ++i) {
-		if(nat_def.values[i] == 0.f)
-			continue;
-
-		auto offset = uint32_t(nat_def.offsets[i].index());
-		auto data = national_modifier_names[offset];
-
+		if(!bool(nat_def.offsets[i]))
+			break;
+		auto data = national_modifier_names[nat_def.offsets[i].index()];
 		auto box = text::open_layout_box(layout, indentation);
 		text::add_to_layout_box(layout, state, box, text::produce_simple_string(state, data.name), text::text_color::white);
 		text::add_to_layout_box(layout, state, box, std::string_view{":"}, text::text_color::white);
 		text::add_space_to_layout_box(layout, state, box);
 		auto color = data.positive_is_green
-		                 ? (nat_def.values[i] > 0.f ? text::text_color::green : text::text_color::red)
-		                 : (nat_def.values[i] > 0.f ? text::text_color::red : text::text_color::green);
+		                 ? (nat_def.values[i] >= 0.f ? text::text_color::green : text::text_color::red)
+		                 : (nat_def.values[i] >= 0.f ? text::text_color::red : text::text_color::green);
+		text::add_to_layout_box(layout, state, box, format_modifier_value(state, nat_def.values[i], data.type), color);
+		text::close_layout_box(layout, box);
+	}
+}
+
+static void acting_modifier_description(sys::state& state, text::layout_base& layout, dcon::modifier_id mid, int32_t indentation, dcon::national_modifier_value nmid, dcon::provincial_modifier_value pmid, float scaled = 1.f) {
+	auto fat_id = dcon::fatten(state.world, mid);
+	const auto& prov_def = fat_id.get_province_values();
+	for(uint32_t i = 0; i < prov_def.modifier_definition_size; ++i) {
+		if(!bool(prov_def.offsets[i]))
+			break;
+		if(prov_def.offsets[i] != pmid)
+			continue;
+		
+		auto data = province_modifier_names[pmid.index()];
+		auto box = text::open_layout_box(layout, indentation);
+		text::add_to_layout_box(layout, state, box, text::produce_simple_string(state, fat_id.get_name()), text::text_color::white);
+		text::add_to_layout_box(layout, state, box, std::string_view{":"}, text::text_color::white);
+		text::add_space_to_layout_box(layout, state, box);
+		auto value = prov_def.values[i] * scaled;
```

Comment:
This is the actual value that you should show, but that is not what you send to the text box. Also, if the scale is 0 the modifier should be skipped completely